### PR TITLE
Setup Integration Testing with basic module-scan/bsd tests

### PR DESCRIPTION
**Pull request by carolinemodic**
Sets up a top-level directory for integration testing with different subfolders that each test a different "bundle" of apps. I'm not entirely sure that this structure will make sense but as we add other bundles we can play with it. 

Initially there is just one bundle for bsd & module-scan, with 1 test file and 2 different test cases to test basic setup and configuration. 

I think it would be cool to test actually "scanning" images in the bsd/module-scan tests but with that needing to be set as an environment variable I wasn't sure the best way to accomplish that. 

Video of current tests: 

https://user-images.githubusercontent.com/14897017/122471991-d6183a00-cf74-11eb-9629-c52f31bc2bb0.mp4

### CODE REVIEW

File: integration-testing/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Integration Testing
+
+Home for integration testing different apps that need to be tested in conjuction with one another. Copy an existing subfolder if you need to set up a new bundle of apps to test.
+
+## App Bundles
+
+bsd - Tests the frontend of bsd with the backend of module-scan
+
+## Development
```
Comment (eventualbuddha) [inline integration-testing/README.md:9]: Should we add a note about installing Google Chrome? I didn't have it and it complained when running `pnpm test:ci`. Running `pnpm test:watch` worked, though.

File: integration-testing/bsd/script/react-scripts [hunk 2]
```diff
@@ -0,0 +1 @@
+../../../script/react-scripts
```
Comment (eventualbuddha) [inline integration-testing/bsd/script/react-scripts:1]: I don't think this script is necessary.

File: integration-testing/bsd/cypress/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    "strict": true,
+    "target": "es2015"
+  },
+  "include": [
+    "../node_modules/cypress",
+    "../node_modules/@testing-library/cypress",
+    "**/*.ts"
+  ],
+	"references": [
+    { "path": "../libs/fixtures" },
```
[the remark this answers is not in the document]
Comment (eventualbuddha) [reply integration-testing/bsd/cypress/tsconfig.json:27]: Also, wrong path:
```suggestion
    { "path": "../../../libs/fixtures" },
```

File: integration-testing/bsd/package.json [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  "private": true,
+  "scripts": {
+    "build": "pnpm install && make build",
+    "eject": "script/react-scripts eject",
+    "format": "prettier '**/*.+(css|graphql|json|less|md|mdx|sass|scss|yaml|yml)' --write",
+    "lint": "eslint . --quiet && pnpm stylelint:run",
+    "lint:fix": "eslint . --fix --quiet && pnpm stylelint:run:fix",
+    "start": "make run",
+    "stylelint:run": "stylelint 'cypress/integration/*.{js,jsx,ts,tsx}' ",
+    "stylelint:run:fix": "stylelint 'cypress/integration/*.{js,jsx,ts,tsx}' --fix",
```
Comment (eventualbuddha) [inline integration-testing/bsd/package.json:13]: These don't appear to be useful as there's no CSS to lint.

File: integration-testing/bsd/package.json [hunk 5]
```diff
@@ -0,0 +1,87 @@
+{
+  "name": "@votingworks/integration-testing-bsd",
+  "version": "0.1.0",
+  "private": true,
+  "scripts": {
+    "build": "pnpm install && make build",
+    "eject": "script/react-scripts eject",
```
Comment (eventualbuddha) [inline integration-testing/bsd/package.json:7]: This isn't a React app, so this isn't needed.

File: integration-testing/bsd/package.json [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+      "not op_mini all"
+    ],
+    "development": [
+      "last 1 chrome version",
+      "last 1 firefox version",
+      "last 1 safari version"
+    ]
+  },
+  "devDependencies": {
+    "@codemod/parser": "^1.0.7",
```
Comment (eventualbuddha) [inline integration-testing/bsd/package.json:52]: Can be removed if stylelint is.

File: integration-testing/bsd/package.json [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+    "eslint-config-prettier": "^6.10.0",
+    "eslint-config-react-app": "^5.2.1",
+    "eslint-import-resolver-node": "^0.3.4",
+    "eslint-plugin-cypress": "^2.11.2",
+    "eslint-plugin-flowtype": "^5.2.0",
+    "eslint-plugin-import": "^2.20.1",
+    "eslint-plugin-jest": "^23.6.0",
+    "eslint-plugin-jsx-a11y": "^6.2.3",
+    "eslint-plugin-prettier": "^3.1.2",
+    "eslint-plugin-react": "^7.21.5",
```
Comment (eventualbuddha) [inline integration-testing/bsd/package.json:71]: Unnecessary.

File: integration-testing/bsd/package.json [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+    "development": [
+      "last 1 chrome version",
+      "last 1 firefox version",
+      "last 1 safari version"
+    ]
+  },
+  "devDependencies": {
+    "@codemod/parser": "^1.0.7",
+    "cypress-file-upload": "^5.0.7",
+    "typescript": "^4.2.0",
```
Comment (eventualbuddha) [inline integration-testing/bsd/package.json:54]: Listed twice? Not sorted?

File: integration-testing/bsd/package.json [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+    "jest-environment-jsdom-sixteen": "^1.0.3",
+    "lint-staged": "^10.0.7",
+    "prettier": "^2.0.5",
+    "sort-package-json": "^1.44.0",
+    "start-server-and-test": "^1.12.5",
+    "stylelint": "^13.1.0",
+    "stylelint-config-palantir": "^4.0.1",
+    "stylelint-config-prettier": "^8.0.1",
+    "stylelint-config-styled-components": "^0.1.1",
+    "stylelint-processor-styled-components": "^1.9.0",
```
Comment (eventualbuddha) [inline integration-testing/bsd/package.json:84]:
```suggestion
```

File: vxsuite.code-workspace [hunk 10]
```diff
@@ -47,6 +47,9 @@
     },
     {
       "path": "libs/utils"
+    },
+    {
+      "path": "integration-testing/bsd"
```
Comment (eventualbuddha) [inline vxsuite.code-workspace:52]: Add a `name` to distinguish it from `bsd` the app?
```suggestion
      "path": "integration-testing/bsd",
      "name": "integration-testing/bsd"
```

File: integration-testing/bsd/package.json [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+      "stylelint",
+      "eslint --quiet --fix"
+    ],
+    "*.css": [
+      "stylelint --config .stylelintrc-css.js --fix"
+    ],
+    "package.json": [
+      "sort-package-json"
+    ]
+  },
```
Comment (eventualbuddha) [inline integration-testing/bsd/package.json:38]: None of this stuff currently works in the monorepo, so maybe just leave it out.

**Review by eventualbuddha — APPROVED** [review]
Needs some changes, but overall approach is sound.

**Comment by carolinemodic** [comment]
I'm working on updating bsd to run with scanning actual ballot images but its proving to be somewhat involved so going to merge this and make a subsequent PR for those changes.

**Pull request merged**